### DISCUSSION TITLE
Trojan detected in gluetun entrypoint file

### DISCUSSION THREAD

**Discussion by wjoshraymond** [qdm12/gluetun#1151] [Q&A] [answered]
Hello folks,

Just want to share how much I love gluetun in my home lab setup. 

I am running into a weird issue with my AV detecting a trojan in the entrypoint file:

```
/share/Container/container-station-data/lib/docker/overlay2/f844eba8ee3cff74850a8a3507e2aa8a8c324b056ccabf7894b67a397a8e1902/diff/gluetun-entrypoint: Unix.Trojan.Agent-6762136-0 FOUND

/share/Container/container-station-data/lib/docker/overlay2/c43bc0a5640d2d140fcb34ef48488339d3e22c1c4c0224cc1f8973d9685f6018/diff/gluetun-entrypoint: Unix.Trojan.Agent-6762136-0 FOUND

/share/Container/container-station-data/lib/docker/overlay2/36ba3d29b3e7bc5cc0f19181bad1b39b378b0aaadc23175cd5efc9fadec25b50/merged/gluetun-entrypoint: Unix.Trojan.Agent-6762136-0 FOUND
```

To provide some background, I run gluetun on my QNAP within Container Station. Recently (last 3-4 weeks) I started receiving the above messages from my AV. The AV uses ClamAV as its virus signature database in case that helps. 

Please let me know if there's any cause for concern or even if this is a known/benign issue and I can silence the alerts on my end. If there's any additional information that would help, also happy to share. 

Thanks again!

Best,
Josh

**Comment by qdm12** [comment]
Hi josh,

Interesting, the Unraid forums also got a report of a trojan and I didn't know what it was. See https://forums.unraid.net/topic/111725-support-diamondprecisioncomputing-all-images-and-files/?do=findComment&comment=1167928

I'm gonna run the binary on my end against virustotal and get back to you. EDIT: https://www.virustotal.com/gui/file/d83792eaed574633ea3dbe27b7252f8710592ddd3b3cba24d24b54949991fbb6?nocache=1 so yeah false positive.

I guess it's most likely a false positive due to it being written in Go (a lot of viruses are written in Go since it's hard to reverse engineer due to its unique compilation) and also containing a bunch of networking in Go (wireguard userspace, ip routing with netlink).

The funny thing is I never coded a trojan or any virus, so I can't 100% tell why it would detect it as such.

**Comment by qdm12** [comment, marked as answer]
See https://github.com/qdm12/gluetun/discussions/1152

**Reply by wjoshraymond** [reply]
Thanks @qdm12 , appreciate your quick feedback and detailed announcement.
